distribution/tools/upgrade-cli/src/main/java/org/opensearch/upgrade/InstallPluginsTask.java

```diff
@@ -28,27 +56,46 @@ public void accept(final Tuple<TaskInput, Terminal> input) {
             return;
         }
         terminal.println("Installing core plugins ...");
-        final ProcessBuilder processBuilder = new ProcessBuilder();
+        List<String> manualPlugins = new ArrayList<>();
+
         for (String plugin : taskInput.getPlugins()) {
-            // TODO - validate if plugin is an official plugin for OpenSearch.
-            final String command = taskInput.getOpenSearchBin().resolve("opensearch-plugin") + " install " + plugin;
-            if (OS.WINDOWS == OS.current()) {
-                processBuilder.command("cmd.exe", "/c", command);
+            if (OFFICIAL_PLUGINS.contains(plugin)) {
+                executeInstallPluginCommand(plugin, taskInput, terminal);
```

Comment:
Noob question here: What are official plugins? Who determines what can go into `plugins.txt` file? Is there any support for 3rd party plugins?
I am trying to understand how the upgrade tool can be used. For example if I manage Elasticsearch cluster 6.x with several 3rd party plugins and want to upgrade to OpenSearch cluster can I benefit from this upgrade tool?

Reply:
The term **official** rather should mean the core plugins which are part of the OpenSearch repository (under the **plugins** directory).  The plugins.txt is created as part of the distribution. There is a [working readme](https://github.com/opensearch-project/OpenSearch/blob/808d577be9cba15bbe4d10634946ae071ccbf3da/distribution/tools/upgrade-cli/README.md) on the upgrade tool.  Currently no plans to add support for 3rd party plugins.